pages/[report]/[fight]/index.tsx

```diff
@@ -1,118 +1,112 @@
-import { GetBossFightsToLog } from '../../../mocks/getBossFightsToLog';
-
-interface IFight {
-  fight: {
-    difficulty: number;
-    name: string;
-    id: number;
-    kill: string;
-    fightPercentage: number;
-    friendlyPlayers: [number];
-  };
-  report: {
-    endTime: number;
-    startTime: number;
-    guild: { name: string; id: number; faction: { name: string } };
-  }
-}
+import Script from 'next/script';
+import { Actions } from '../../../interfaces';
+import ListOfPlayerRoles from '../../../components/player/listOfPlayerRoles';
+import { IFightResponse, IPlayerDetails } from '../../../interfaces/FightResponse';
+import { fetchFightData } from './fetchDataToFight';
+import { FEATURE_IDS } from '../../../constants/FEATURE_IDS';
+import GearIssues from '../../../features/gearIssues';
+import { IChoice } from '../../../interfaces/Choice';
 
 const Main = styled.div`
-  padding: 4rem 0;
   flex: 1;
-  display: flex;
-  flex-direction: column;
+  padding: 5rem;
+  @media (max-width: 960px) {
+    padding: 2rem;
+  }
 `;
 
-const CharacterContainer = styled.div`
+const ContentContainer = styled.div`
   display: flex;
-  flex-direction: column;
-  margin: 0 4rem 0 4rem;
-`;
+  flex-direction: row;
 
-const CharacterInfo = styled.div`
-  margin: 0.5rem 0 0.5rem 0;
-  cursor: pointer;
+  @media (max-width: 1200px) {
+    flex-direction: column;
+  }
 `;
 
-const EncounterContainer = styled.div`
-  display: flex;
-  flex-direction: row;
-  justify-content: space-evenly;
-  margin-bottom: 2rem;
+const Content = styled.div`
+  margin: 0 1rem 1rem 0;
 `;
 
-interface IUpdateQueryParam {
-  id: string,
-  value: number;
-}
+export const getServerSideProps: GetServerSideProps = async (props) => {
+  const { query } = props;
+  const action: Actions = 'FIGHT';
+  console.log('process: ', `${process.env.BACKEND_URL}?action=${action}&code=${query?.report}&fight=${query?.fight}&encounterID=${query?.encounterID}&startTime=${query?.startTime}&endTime=${query?.endTime}`);
+  const { data } = await axios.get(
+    `${process.env.BACKEND_URL}?action=${action}&code=${query?.report}&fight=${query?.fight}&encounterID=${query?.encounterID}&startTime=${query?.startTime}&endTime=${query?.endTime}`,
+  );
+  const {
+    player,
+    guild,
+  }: IFightResponse = data;
+  return {
+    props: {
+      player,
+      guild,
+    },
+  };
+};
 
-const Fight = ({ fight, report }: IFight) => {
-  console.log('loggi ', fight);
+const Fight = (fightResponse: IFightResponse) => {
+  const [fightData, setFightData] = useState(fightResponse);
+  const [player, setSelectedPlayer] = useState<IPlayerDetails | null>(null);
+  const [choice, setChoice] = useState<IChoice>(null);
   const router = useRouter();
+  const { query } = router;
+  const action:Actions = 'FEATURE_GEAR_ISSUES';
+  const dataQuery = `?action=${action}&code=${query?.report}&fight=${query?.fight}&encounterID=${query?.encounterID}&startTime=${query?.startTime}&endTime=${query?.endTime}`;
```

Comment:
These should be calculated only when the addressed values change:
```suggestion
  const action:Actions = useMemo(() => 'FEATURE_GEAR_ISSUES', []);
  const dataQuery = useMemo(() => `?action=${action}&code=${query?.report}&fight=${query?.fight}&encounterID=${query?.encounterID}&startTime=${query?.startTime}&endTime=${query?.endTime}`, [action, query]);
```

also axios is able to generate the query paramas by adding a json for the params key in the axios config like: 
```
params: {
action,
code: query?.report,
fight: query?.fight,
encounterID: query?.encounterID,
startTime: query?.startTime,
endTime: query?.endTime
}
```